File: IOS/AttendanceCore/Sources/AttendanceCore/AttendancePalette.swift
#if os(iOS)
import SwiftUI

public enum AttendancePalette {
    public static var background: Color {
        Color(uiColor: UIColor { traits in
            traits.userInterfaceStyle == .dark ? .black : .white
        })
    }

    public static var foreground: Color {
        Color(uiColor: UIColor { traits in
            traits.userInterfaceStyle == .dark ? .white : .black
        })
    }

    public static var secondaryForeground: Color {
        Color(uiColor: UIColor { traits in
            traits.userInterfaceStyle == .dark
                ? UIColor.white.withAlphaComponent(0.65)
                : UIColor.black.withAlphaComponent(0.55)
        })
    }

    public static var lowAttendance: Color {
        .red
    }

    public static func rowColor(for percentage: Double) -> Color {
        percentage < 70 ? lowAttendance : foreground
    }
}
#endif

File: Android/app/app/src/main/kotlin/edu/bmu/attendance/ui/AttendancePalette.kt
package edu.bmu.attendance.ui

import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color

object AttendancePalette {
    @Composable
    fun background(): Color = MaterialTheme.colorScheme.background

    @Composable
    fun sectionBackground(): Color = MaterialTheme.colorScheme.surfaceContainerHigh

    @Composable
    fun foreground(): Color = MaterialTheme.colorScheme.onSurface

    @Composable
    fun secondaryForeground(): Color = MaterialTheme.colorScheme.onSurfaceVariant

    val lowAttendance: Color = Color.Red

    @Composable
    fun rowColor(percentage: Double): Color =
        if (percentage < 70.0) lowAttendance else MaterialTheme.colorScheme.onSurface

    val statusOk: Color = Color(0xFF34C759)
    val statusFailed: Color = Color.Red
    val actionBlue: Color = Color(0xFF0A84FF)
}
